### ISSUE TITLE
Acting strange

### ISSUE THREAD

**Issue by lorvent** [REBELinBLUE/deployer#233]
Before submitting your issue, please make sure that you've checked all of the checkboxes below.

- [x] You're running the [latest release](https://github.com/REBElinBLUE/deployer/releases/latest) version of Deployer.
- [x] Ensure that you're running at least PHP 5.5.9, you can check this by running `php -v`
- [x] You've ran `composer install --no-dev` from the root of your installation.
- [x] You've ran `npm install --production` from the root of your installation.
- [x] You've ran `rm -rf bootstrap/cache/*` from the root of your installation.
- [ ] You have restarted the queue listener and node socket server.

To help us better understand your issue, please answer the following.

### Expected behaviour

Deployment should be done from top to bottom
### Actual behaviour

below pic explains it better
![x](http://www.clipular.com/c/4888928164773888.png?k=A4y-BmCHM3AliiSa0XO7bCo0ENE)

and sometimes first it displays success message with Deployment finished then after few seconds, it displays another message saying Deployment failed.

### Steps to reproduce

*If your issue requires any specific steps to reproduce, please outline them here.*

### Environment info
Operating System:
PHP Version:
Node Version:
Database System:
Database Version:

### Logs (see storage/logs/) or other output that would be helpful
(If logs are large, please upload as attachment).

**Comment by REBELinBLUE** [comment]
Ah, I thought I'd caught this. It is nothing to worry about, the deployment is done in order but occasionally the browser doesn't receive the event to update the UI for the running task

**Comment by lorvent** [comment]
I am seeing this more and more for beanstalkd compared to redis

it happens like this

* all steps top to bottom finished
* displays deployment finished message at bottom
* suddenly some step like a custom defined will pick up
* again it displays deployment finished message at bottom

and i get 2-3 slack notifications

**Comment by REBELinBLUE** [comment]
Hmm, it sounds almost like the jobs are being put make into the queue. What is your supervisor config like? 

Stephen Ball
stephen@rebelinblue.com

**Comment by lorvent** [comment]
its like in docs

`$ cat /etc/supervisor/supervisord.conf` produces

```
; supervisor config file

[unix_http_server]
file=/var/run/supervisor.sock   ; (the path to the socket file)
chmod=0700                       ; sockef file mode (default 0700)

[supervisord]
logfile=/var/log/supervisor/supervisord.log ; (main log file;default $CWD/supervisord.log)
pidfile=/var/run/supervisord.pid ; (supervisord pidfile;default supervisord.pid)
childlogdir=/var/log/supervisor            ; ('AUTO' child log dir, default $TEMP)

; the below section must remain in the config file for RPC
; (supervisorctl/web interface) to work, additional interfaces may be
; added by defining them in separate rpcinterface: sections
[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[supervisorctl]
serverurl=unix:///var/run/supervisor.sock ; use a unix:// URL  for a unix socket

; The [include] section can just contain the "files" setting.  This
; setting can list multiple files (separated by whitespace or
; newlines).  It can also contain wildcards.  The filenames are
; interpreted as relative to this file.  Included files *cannot*
; include files themselves.

[include]
files = /etc/supervisor/conf.d/*.conf

[program:deployer]
command=php artisan queue:work --queue=deployer-high,deployer-low --sleep=3 --tries=1 --daemon
directory=/var/www/html/deploy
process_name=queue_%(process_num)s
numprocs=3
numprocs_start=0
stdout_logfile=/var/www/html/deploy/storage/logs/supervisord-%(process_num)s-stdout.log
stderr_logfile=/var/www/html/deploy/storage/logs/supervisord-%(process_num)s-stderr.log
stderr_logfile_maxbytes=1MB
stdout_logfile_maxbytes=1MB
redirect_stderr=true
autostart=true
autorestart=true

[program:deployer-socketio]
command=node socket.js
directory=/var/www/html/deploy
numprocs=1
autostart=true
autorestart=true
environment=NODE_ENV=production
stderr_logfile=/var/www/html/deploy/storage/logs/node-stderr.log
stdout_logfile=/var/www/html/deploy/storage/logs/node-stdout.log
stderr_logfile_maxbytes=1MB
stdout_logfile_maxbytes=1MB

[program:deployer-broadcast]
command=php artisan queue:work --queue=deployer-default --sleep=3 --tries=1 --daemon
directory=/var/www/html/deploy
numprocs=1
stderr_logfile=/var/www/html/deploy/storage/logs/broadcast-stderr.log
stdout_logfile=/var/www/html/deploy/storage/logs/broadcast-stdout.log
stderr_logfile_maxbytes=1MB
stdout_logfile_maxbytes=1MB
redirect_stderr=true
autostart=true
autorestart=true
```

**Comment by REBELinBLUE** [comment]
Interesting. Looking into it but it might take a while to figure out. All the queue stuff if handled by laravel so there shouldn't be any difference between the 2 unless there is a bug in laravel

**Comment by REBELinBLUE** [comment]
Yep getting the same thing when using redis, very odd

**Comment by REBELinBLUE** [comment]
https://github.com/laravel/framework/issues/8577 hmm, looks like it could be an issue

**Comment by REBELinBLUE** [comment]
It seems like it is an issue with Laravel, which may be fixed in 5.3.

**Comment by lorvent** [comment]
yeah...looks like a bug with laravel itself for ages.
